app/routes/user/password.tsx

```diff
@@ -0,0 +1,140 @@
+    return /^\s|\s$/.test(str)
+  }
+  const [newPassword, setNewPassword] = useState('')
+  const [confirmPassword, setConfirmPassword] = useState('')
+  const [isNewPasswordTouched, setIsNewPasswordTouched] = useState(false)
+  const [isConfirmPasswordTouched, setIsConfirmPasswordTouched] =
+    useState(false)
+  const passwordsMatch = newPassword === confirmPassword
+  const passwordsAreEmpty = !newPassword && !confirmPassword
+  const shouldDisableSubmit =
+    passwordsAreEmpty || !isNewPasswordTouched || !passwordsMatch
+  const isMatchError =
+    !isConfirmPasswordTouched || passwordsMatch ? '' : 'usa-input--error'
+
+  const valid =
+    !startsOrEndsWithWhitespace(newPassword) && checkPassword(newPassword)
+  const isValidPassword =
+    !isNewPasswordTouched || valid ? '' : 'usa-input--error'
+
+  return (
+    <>
+      <h1>Reset Password</h1>
+      <>
+        <fetcher.Form method="POST">
+          <Label htmlFor="oldPassword">Old Password</Label>
+          <TextInput
+            data-focus
+            name="oldPassword"
+            id="oldPassword"
+            type="password"
+            placeholder="Old Password"
+          />
+          <Label htmlFor="newPassword">New Password</Label>
+          <TextInput
+            className={isValidPassword}
+            name="newPassword"
+            id="newPassword"
+            type="password"
+            placeholder="New Password"
+            onChange={(e) => {
+              setNewPassword(e.target.value)
+              setIsNewPasswordTouched(true)
+            }}
+            value={newPassword}
+          />
+          <Label htmlFor="confirmPassword">Retype New Password</Label>
+          <TextInput
+            className={isMatchError}
+            name="confirmPassword"
+            id="confirmPassword"
+            type="password"
+            placeholder="Retype New Password"
+            onChange={(e) => {
+              setConfirmPassword(e.target.value)
+              setIsNewPasswordTouched(true)
+              setIsConfirmPasswordTouched(true)
+            }}
+            value={confirmPassword}
+          />
+
+          <Button
+            className="margin-y-2"
+            disabled={shouldDisableSubmit}
+            type="submit"
+          >
+            Reset Password
+          </Button>
+          {fetcher.state !== 'idle' && (
+            <>
+              <Spinner /> Saving...
+            </>
+          )}
+          {fetcher.state === 'idle' && fetcher.data !== undefined && (
+            <>
+              <Icon.Check color="green" /> Saved
+            </>
+          )}
+        </fetcher.Form>
+      </>
+      <h3 className="site-preview-heading margin-0">
+        New password must contain:
+      </h3>
+      <ul className="usa-list">
+        <li>A lower case letter</li>
+        <li>An upper case letter</li>
+        <li>A number</li>
+        <li>At least 8 characters</li>
+        <li>At least 1 special character or space</li>
```

Comment:
What is a "special character" here?

Reply:
typically anything that is not A-z or 0-9. This IS a configurable value though, so it depends on what we have configured or if it the default. I believe the default is ^ $ * . [ ] { } ( ) ? " ! @ # % & / \ , > < ' : ; | _ ~ ` = + - 
as [defined in the developers guide](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-settings-policies.html)
I was unable to pull the password policy with my credentials, the connection was refused. So I do not know if we have altered the special character definition.

Reply:
It's set to the default:

```
            "PasswordPolicy": {
                "MinimumLength": 8,
                "RequireUppercase": true,
                "RequireLowercase": true,
                "RequireNumbers": true,
                "RequireSymbols": true,
                "TemporaryPasswordValidityDays": 7
            }
```